fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/SumLiteralRewrite.java

```diff
@@ -0,0 +1,157 @@
+                    newAggOutput.add(namedSum);
+                    newAggOutput.add(namedCount);
+                });
+        LogicalAggregate<?> newAgg = agg.withAggOutput(ImmutableList.copyOf(newAggOutput));
+        List<NamedExpression> newProjects = agg.getOutputExpressions().stream()
+                .map(namedExpr -> {
+                    if (!funcMap.containsKey(namedExpr)) {
+                        return namedExpr.toSlot();
+                    }
+                    Expression originExpr = funcMap.get(namedExpr).first;
+                    Literal literal = funcMap.get(namedExpr).second;
+                    if (namedExpr.child(0).child(0) instanceof Add) {
+                        Expression newExpr = new Add(exprToSum.get(originExpr),
+                                new Multiply(literal, exprToCount.get(originExpr)));
+                        return new Alias(namedExpr.getExprId(), newExpr, namedExpr.getName());
+                    } else {
+                        Expression newExpr = new Subtract(exprToSum.get(originExpr),
+                                new Multiply(literal, exprToCount.get(originExpr)));
+                        return new Alias(namedExpr.getExprId(), newExpr, namedExpr.getName());
+                    }
+                }).collect(Collectors.toList());
+
+        return new LogicalProject<>(newProjects, newAgg);
+    }
+
+    private NamedExpression constructSum(Expression child, Map<AggregateFunction, NamedExpression> existedAggFunc) {
+        Sum sum = new Sum(child);
+        NamedExpression namedSum;
+        if (existedAggFunc.containsKey(sum)) {
+            namedSum = existedAggFunc.get(sum);
+        } else {
+            namedSum = new Alias(sum);
+        }
+        return namedSum;
+    }
+
+    private NamedExpression constructCount(Expression child, Map<AggregateFunction, NamedExpression> existedAggFunc) {
+        Count count = new Count(child);
+        NamedExpression namedCount;
+        if (existedAggFunc.containsKey(count)) {
+            namedCount = existedAggFunc.get(count);
+        } else {
+            namedCount = new Alias(count);
+        }
+        return namedCount;
+    }
+
+    private @Nullable Pair<NamedExpression, Pair<Expression, Literal>> extractSumLiteral(
+            NamedExpression namedExpression) {
+        if (namedExpression.children().size() != 1) {
+            return null;
+        }
+        Expression func = namedExpression.child(0);
+        if (!(func instanceof Sum)) {
+            return null;
+        }
+        Expression child = func.child(0);
+        if (!(child instanceof Add) && !(child instanceof Subtract)) {
+            return null;
+        }
+
+        Expression left = ((BinaryArithmetic) child).left();
+        Expression right = ((BinaryArithmetic) child).right();
+        if (!right.isLiteral() && !(left instanceof Slot)) {
+            // right now, only support slot +/- literal
+            return null;
+        }
```

Comment:
please test decimal type carefully to ensure we can always get correct precision and scale